UnicornEngine/video/source/Camera.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/Camera.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+Camera::Camera(glm::vec3 postion, glm::vec3 direction)
+    : m_aspect(0), m_camPosition(postion), m_upVector(glm::vec3(0.0f, -1.0f, 0.0f)), m_camDirection(direction), m_fov(45.0f), m_znear(0.1f), m_zfar(1000.0f), m_projectionType(ProjectionType::Perspective)
+{
+    UpdateViewMatrix();
+    UpdateProjectionMatrix();
+}
+
+void Camera::SetPerspective(float fov, float aspect, float znear, float zfar)
+{
+    m_fov = fov;
+    m_znear = znear;
+    m_zfar = zfar;
+    m_aspect = aspect;
+    UpdateProjectionMatrix();
+}
+
+void Camera::UpdateAspectRatio(float aspect)
+{
+    m_aspect = aspect;
+    UpdateProjectionMatrix();
+}
+
+void Camera::Translate(const glm::vec3& delta)
+{
+    m_camPosition += delta;
+    UpdateViewMatrix();
```

Comment:
Do you really want to update it on every call or setting some `isDirty` flag and actually updating the matrix on some getters would be better?

For example a chain of `Translate -> Change direction -> Change up vector` right now results in 3 calculations where it could be just 1 whenever the camera is being used for rendering